AndriiZaimak/homework-03/src/middlewares/auth.middleware.js

```diff
@@ -0,0 +1,27 @@
+import jwt from 'jsonwebtoken';
+
+import { UnAuthorizedError } from '#helpers/errors.js';
+
+const { JWT_SECRET } = process.env;
+
+const auth = (req, _res, next) => {
+  const { authorization } = req.headers;
+
+  if (!authorization) throw new UnAuthorizedError('No token provided');
+
+  const [, token] = authorization.split('Bearer ');
+
+  if (!token) throw new UnAuthorizedError('Invalid token format');
+
+  try {
+    const payload = jwt.verify(token, JWT_SECRET);
```

Comment:
try to turn this into promise to unblock other processes during token verification, this is called "to promisify"